### PR TITLE
Feature/debug mode

### PR DESCRIPTION
**Pull request by 902seanryan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,20 +162,41 @@ let promiseData = new Promise(function(resolve, reject) {
...
+```
+By default (above method) it will print a message outlining whether the bellhop instance was a
+child or parent, whether the message was sent or received, and the contents of the message. If you require additional or custom logging you can also pass a function as the flag.
+```javascript
+const log = () => {console.log('Hello World!');}
+bellhop.debug = log; // Hello World!
+```
+If you pass a function to debug three parameters* are passed to help fill out the log statements if required:
+```javascript
+const log = ({isChild, received, message}){
```
Comment (JonGardner-RedSpace) [inline README.md]: Super minor
```suggestion
const log = ({isChild, received, message}) => {
```

File: src/Bellhop.js [hunk 2]
```diff
@@ -99,6 +109,12 @@ export class Bellhop extends BellhopEventDispatcher {
       return;
     }
 
+    if (this.debug && typeof this.debug === 'function') {
```
Comment (JonGardner-RedSpace) [inline src/Bellhop.js]: Maybe extract this to a function call? 
Just in case there's an update later, I noticed it's duplicated down below.

**Comment by JonGardner-RedSpace** [comment]
There's a  `console.error` in the file, should that be routed though `debug` if debug is a function?
@chipbell4

File: README.md (hunk 1 above)
Comment (902seanryan) [reply README.md]: Good catch

File: src/Bellhop.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ export class Bellhop extends BellhopEventDispatcher {
    * Creates an instance of Bellhop.
    * @memberof Bellhop
    * @param { string | number } id the id of the Bellhop instance
+   * @param { boolean | function } debug whether debug mode should be turned on for the bellhop instance
```
Comment (chipbell4) [inline src/Bellhop.js]: This doc block isn't needed because there's no associated parameter.

File: src/Bellhop.js [hunk 4]
```diff
@@ -262,31 +277,44 @@ export class Bellhop extends BellhopEventDispatcher {
...
+  /**
+   * Send either the default log message or the callback provided if debug
+   * is enabled
+   * @method logDebugMessage
+   */
+  logDebugMessage(received = false, message) {
+    if (this.debug && typeof this.debug === 'function') {
+      this.debug({ isChild: this.isChild, received: false, message: message });
+    } else if (this.debug) {
+      console.log(`Bellhop Instance (${this.isChild ? 'Child' : 'Parent'}) ${received ? 'Receieved' : 'Sent'}`, message);
```
Comment (chipbell4) [inline src/Bellhop.js]: I was going to suggest making this a `console.debug` call instead, but it looks like [support is a tad patchy](https://caniuse.com/#feat=mdn-api_console_debug) so let's leave it as is 👍

**Pull request merged**